sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/FileSmbPropertiesHelper.java

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.file.share;
+
+import com.azure.core.http.HttpHeaders;
+
+/**
+ * A class used to allow access to the FileSmb constructor in other packages.
+ */
+public class FileSmbPropertiesHelper {
```

Comment:
This needs to use the accessor pattern seen in implementation, this is still adding a public API change. See the Blob Storage accesshelpers package for examples: https://github.com/Azure/azure-sdk-for-java/tree/main/sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/accesshelpers